FieldworkAIDriver.lua

```diff
@@ -555,57 +558,30 @@ function FieldworkAIDriver:updateFieldworkOffset()
 		self.vehicle.cp.toolOffsetZ + self.aiDriverOffsetZ)
 end
 
-function FieldworkAIDriver:hasSameCourse(otherVehicle)
-	if otherVehicle.cp.driver and otherVehicle.cp.driver.fieldworkCourse then
-		return self.fieldworkCourse:equals(otherVehicle.cp.driver.fieldworkCourse)
-	else
-		return false
-	end
-end
-
---- When working in a group (convoy), do I have to hold so I don't get too close to the
--- other vehicles in front of me?
+--- When working in a group (convoy), 
+--    do I have to hold so I don't get too close to the leading vehicle?
+--    do I have to hold so I don't get too far from the trailing vehicle?
 function FieldworkAIDriver:manageConvoy()
 	if not self.vehicle.cp.convoyActive then return false end
-	--get my position in convoy and look for the closest combine
-	local position = 1
-	local total = 1
-	local closestDistance = math.huge
-	for _, otherVehicle in pairs(CpManager.activeCoursePlayers) do
-		if otherVehicle ~= self.vehicle and otherVehicle.cp.convoyActive and self:hasSameCourse(otherVehicle) then
-			local myWpIndex = self.ppc:getCurrentWaypointIx()
-			local otherVehicleWpIndex = otherVehicle.cp.ppc:getCurrentWaypointIx()
-			total = total + 1
-			if myWpIndex < otherVehicleWpIndex then
-				position = position + 1
-				local distance = (otherVehicleWpIndex - myWpIndex) * courseGenerator.waypointDistance
-				if distance < closestDistance then
-					closestDistance = distance
-				end
-			end
-		end
-	end
-
-	-- stop when I'm too close to the combine in front of me
-	if position > 1 then
-		if closestDistance < self.vehicle.cp.convoy.minDistance then
-			self:debugSparse('too close (%.1f) to other vehicles in group, holding.', closestDistance)
-			self:setSpeed(0)
-		end
-	else
-		closestDistance = 0
-	end
-
-	-- TODO: check for change should be handled by setCpVar()
-	if self.vehicle.cp.convoy.distance ~= closestDistance then
-		self.vehicle:setCpVar('convoy.distance',closestDistance)
-	end
-	if self.vehicle.cp.convoy.number ~= position then
-		self.vehicle:setCpVar('convoy.number',position)
-	end
-	if self.vehicle.cp.convoy.members ~= total then
-		self.vehicle:setCpVar('convoy.members',total)
-	end
+	
+    -- Stop if we're too close to the leading vehicle
+    if self.vehicle.cp.convoy.number > 1 then
+        --self.vehicle.cp.convoy.leadDist = courseplay:nodeToNodeDistance(self.vehicle.rootNode, CpManager.convoys[self.vehicle.cp.convoy.id][self.vehicle.cp.convoy.number-1].rootNode)
+        self.vehicle:setCpVar('convoy.leadDist',courseplay:nodeToNodeDistance(self.vehicle.rootNode, CpManager.convoys[self.vehicle.cp.convoy.id][self.vehicle.cp.convoy.number-1].rootNode))
```

Comment:
This is more of a test for me to see if it behaved properly instead of WP distance, as I've had problems on circle islands and headland manoeuvres. 
I'd prefer using the collision detector you've built paired with WP distance keeping, but I haven't figured out yet how to enable your detector on FieldworkAIDriver.

As for the other remarks, I'll rebase and get on with them, hopefully soon.

Reply:
You can PM me if you have any questions (github@vaiko.us)

Reply:
Did some work here. Please check and advise.